src/Psalm/CallMap.php

```diff
@@ -242,17 +242,17 @@
 'array_pop' => ['mixed', '&rw_stack'=>'array'],
 'array_product' => ['int|float', 'input'=>'array'],
 'array_push' => ['int', '&rw_stack'=>'array', 'var'=>'mixed', '...vars='=>'mixed'],
-'array_rand' => ['int|string|int[]|string[]', 'input'=>'array', 'num_req'=>'int'],
+'array_rand' => ['int|string|array<int,int>|array<int,string>', 'input'=>'array', 'num_req'=>'int'],
 'array_rand\'1' => ['int|string', 'input'=>'array'],
 'array_reduce' => ['mixed', 'input'=>'array', 'callback'=>'callable', 'initial='=>'mixed'],
 'array_replace' => ['array', 'arr1'=>'array', 'arr2'=>'array', '...='=>'array'],
 'array_replace_recursive' => ['array', 'arr1'=>'array', 'arr2'=>'array', '...='=>'array'],
 'array_reverse' => ['array', 'input'=>'array', 'preserve='=>'bool'],
-'array_search' => ['mixed', 'needle'=>'mixed', 'haystack'=>'array', 'strict='=>'bool'],
+'array_search' => ['int|string|false', 'needle'=>'mixed', 'haystack'=>'array', 'strict='=>'bool'],
 'array_shift' => ['mixed', '&rw_stack'=>'array'],
 'array_slice' => ['array', 'input'=>'array', 'offset'=>'int', 'length='=>'?int', 'preserve_keys='=>'bool'],
 'array_splice' => ['array', '&rw_input'=>'array', 'offset'=>'int', 'length='=>'int', 'replacement='=>'array|string'],
-'array_sum' => ['numeric', 'input'=>'array'],
+'array_sum' => ['int|float', 'input'=>'array'],
```

Comment:
I'm on the fence about this too, but I could also override very easily in `FunctionChecker::getArrayReturnType` for when we know the array type args.